### ISSUE TITLE
db corruption errors

### ISSUE THREAD

**Issue by sandeep-sporttotal** [nitrite/nitrite-java#781]
Hello,

We are getting DB corruption error after updating targetSDK of our app to android-32. I realized that this error has been reported previously as well and thus I'm not sure we are able to have permanent fix yet.  I've furnished some relevant logs for now and I can share more details if needed. I would be grateful to get some pointers how to resolve this and also seek some insights why this is an issue after updating targetSDk, because it has been doing good before that.

- Happens only on production signed builds. Debug builds are not seeing this issues.
- Only changes we made is updating targetSdk to 32
- Current nitrite version : 4.0.0-SNAPSHOT
- Error logs :
 
  ```
 Caused by: java.lang.IllegalStateException: Unable to read the page at position 274878041998 [1.4.200/6]
        at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:1)
        at org.h2.mvstore.MVStore.readPage(MVStore.java:6)
        at org.h2.mvstore.MVMap.readPage(MVMap.java:1)
        at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:1)
        at org.h2.mvstore.MVMap.setRootPos(MVMap.java:1)
        at org.h2.mvstore.MVStore.openMap(MVStore.java:23)
        at org.h2.mvstore.MVStore.openMap(MVStore.java:3)
        at org.h2.mvstore.MVStore.openMap(MVStore.java:1)
        at xe.f.D(NitriteMVStore.java:3)
        at fe.f.f(NitriteDatabase.java:1)
        at fe.f.<init>(NitriteDatabase.java:17)
```

**Comment by anidotnet** [comment]
This generally boils down to an issue with previous h2 mvstore 1.4.200. The current development branch is at 4.1.0-SNAPSHOT with h2 mvstore version 2.1.214. Can you please try 4.1.0-SNAPSHOT and make sure to apply all proguard rules (https://github.com/nitrite/nitrite-java/issues/697#issuecomment-1053895919).

Meantime, would you provide some contextual information along with logs? What you were trying to execute when this error happened? A minimum reproducible code would be highly appreciated.

**Comment by sandeep-sporttotal** [comment]
Primarily I upgraded my targetSdk to 31 when I observed this error (No changes in proguard
). Let's call it v2 and as previously my app was targeting below 31, let's call it v1.

It works good when it is a fresh install of either v1 or v2 but it fails when v2 is installed as an upgrade over v1 - which already have existing file-based DB and thus it tries to access the Nitrite object using existing file and not creating new.  This is the time it throws error as I decribed in my post and will not recover until I delete the existing file and recreate it.

```
Nitrite.builder()
            .loadModule(
                MVStoreModule.withConfig()
                    .filePath(dbFile.absolutePath)
                    .compress(true)
                    .build()
            )
            .loadModule(JacksonMapperModule())
            .openOrCreate(..., ....)
```
